src/commons/editor/Comments.tsx

```diff
@@ -0,0 +1,362 @@
+        } else {
+          removeCommentEdit(comment);
+        }
+      }
+    };
+  }
+
+  function confirmSubmit(comment: IComment) {
+    return async function (evt: any) {
+      // TODO: figure out a method for edits
+      const newComment: IComment = {
+        ...comment,
+        datetime: Date.now()
+      };
+
+      // This is to give a spinner to the users.
+      setLoadingStatus({
+        ...loadingStatus,
+        [comment.id]: true
+      });
+
+      try {
+        await sendToServer(newComment); // TODO: STUB FUNCTION, PLEASE UPDATE.
+        updateComment(newComment);
+        removeCommentEdit(newComment);
+        setErrorMsgs(omit(errorMsgs, [comment.id]));
+      } catch (e) {
+        setErrorMsgs({
+          ...errorMsgsRef.current,
+          [comment.id]: e
+        });
+      } finally {
+        // Pop quiz: why do we need to ref.current?
+        setLoadingStatus(omit(loadingStatusRef.current, [comment.id]));
+      }
+    };
+  }
+
+  function editComment(comment: IComment) {
+    return function (evt: any) {
+      // Puts this comment as being updated
+      updateCommentsBeingEdited(comment);
+    };
+  }
+
+  function deleteComment(comment: IComment) {
+    return async function (evt: any) {
+      try {
+        await sendToServer(comment);
+        removeComment(comment);
+        setErrorMsgs(omit(errorMsgs, [comment.id]));
+      } catch (e) {
+        setErrorMsgs({
+          ...errorMsgsRef.current,
+          [comment.id]: e
+        });
+      }
+    };
+  }
+
+  function setCollapse(comment: IComment, status: boolean) {
+    return function (evt: any) {
+      updateComment({
+        ...comment,
+        isCollapsed: status
+      });
+    };
+  }
+
+  function setCollapseAll(status: boolean) {
+    return function (evt: any) {
+      const newComments = comments.map(comment => ({
+        ...comment,
+        isCollapsed: status
+      }));
+      updateComment(...newComments);
+    };
+  }
+
+  // ----------------- RENDERING -----------------
+  const isCollapsed = some(comments, c => c.isCollapsed);
+  return (
+    <div className="comments-container" style={commentsContainerStyles}>
+      <div
+        className="gutter-controls"
+        style={{ float: 'left' }}
+        onClick={setCollapseAll(!isCollapsed)}
+      >
+        <Icon icon={isCollapsed ? 'small-plus' : 'small-minus'}></Icon>
+      </div>
+      {comments.map(comment => {
+        const { id, isCollapsed } = comment; // Only the main comment is collapsed.
+        const error = errorMsgs[id] || '';
+        const isLoading = loadingStatus[id] || false;
+        const displayComment = commentsBeingEdited[id] ? commentsBeingEdited[id] : comment;
+        const { profilePic, username, text, datetime } = displayComment;
+        const isEditing: boolean = !!commentsBeingEdited[id];
+        if (isCollapsed) {
+          return (
+            <div
+              onClick={setCollapse(comment, false)}
+              style={{
```

Comment:
We can probably assign classes to these elements and pull out all the inline style into a separate CSS file just to make it look less cluttered. Low-priority thing.

Reply:
Yeah i left it in because prototype. If got major changes then easier than digging thru css conflicts.